h/groups/services.py

```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
+
+from h.accounts import get_user
+from h.models import Group
+
+
+class GroupsService(object):
+
+    """A service for manipulating groups and group membership."""
+
+    def __init__(self, session, user_fetcher):
+        self.session = session
+        self.user_fetcher = user_fetcher
+
+    def create(self, name, userid):
+        """
+        Create a new group.
+
+        :param name: the human-readable name of the group
+        :param userid: the userid of the group creator
+
+        :returns: the created group
+        """
+        creator = self.user_fetcher(userid)
+        group = Group(name=name, creator=creator)
+        self.session.add(group)
+        self.session.flush()
+
+        return group
+
+    def member_join(self, group, userid):
+        """Add `userid` to the member list of `group`."""
```

Comment:
Single-backticks is actually italics in rST, it's double backticks for code spans. Not sure if that's intentional?